csp-mixin/signals/virtualMachines.libsonnet

```
@@ -0,0 +1,141 @@
+local commonlib = import 'common-lib/common/main.libsonnet';
+function(this)
+  {
+    local s = self,
+    filteringSelector: this.filteringSelector,
+    groupLabels: this.groupLabels,
+    instanceLabels: this.instanceLabels,
+    aggLevel: 'instance',
```

```suggestion
    filteringSelector: this.filteringSelector,
    groupLabels: this.groupLabels,
    instanceLabels: this.instanceLabels,
    aggLevel: 'instance',
    aggKeepLabels: ['resourceName', 'resourceGroup', 'subscriptionName'],
```

nit: if you change all types in this group from 'raw' to 'gauge','counter', 'histogram' etc, then you can actually set aggKeepLabels here instead of defining it in every signal. Any automatic aggregations would include those common labels then.

---

more on this https://github.com/grafana/jsonnet-libs/blob/master/common-lib/common/signal/README.md